httprouter/jsonrpcapi/jsonrpcapi.go

```diff
@@ -76,33 +76,49 @@ func (s *SignedJRPC) AllowPrivate(allow bool) {
 }
 
 // AuthorizeRequest implements the httprouter interface
-func (s *SignedJRPC) AuthorizeRequest(data interface{}, isAdmin bool) (bool, error) {
+func (s *SignedJRPC) AuthorizeRequest(data interface{},
+	accessType httprouter.AuthAccessType) (bool, error) {
 	request, ok := data.(*SignedJRPCdata)
 	if !ok {
 		panic("type is not SignedJRPCdata")
 	}
 	// If admin method, compare with the admin address
-	if isAdmin {
+	switch accessType {
+	case httprouter.AccessTypeAdmin:
 		if s.adminAddr == nil {
 			return false, s.returnError("admin address not defined", request.ID)
 		}
 		if *s.adminAddr != request.Address {
 			return false, s.returnError("admin address do not match", request.ID)
 		}
 		return true, nil
-	}
-	// If private method, check authentication.
-	// If no authorized addresses configured, allows any address.
-	if s.allowPrivate && request.Private {
-		if len(s.signer.Authorized) == 0 {
-			return true, nil
+	case httprouter.AccessTypeQuota:
```

Comment:
This case should throw a panic() IMO because it means the developer is not aware of this restriction and panic is the best way to identify were exactly the error comes from.